disnake/appinfo.py

```diff
@@ -402,6 +448,156 @@ def user_install_type_config(self) -> Optional[InstallTypeConfiguration]:
+
+            It's recommended to use :attr:`guild_install_type_config` and :attr:`user_install_type_config`
+            instead of this parameter, as this parameter is soft-deprecated by Discord.
+        custom_install_url: Optional[:class:`str`]
+            The custom installation url for this application.
+        role_connections_verification_url: Optional[:class:`str`]
+            The application's role connection verification entry point,
+            which when configured will render the app as a verification method
+            in the guild role verification configuration.
+        interactions_endpoint_url: Optional[:class:`str`]
+            The application's interactions endpoint URL.
+        icon: Optional[|resource_type|]
+            Update the application's icon asset, if any.
+        cover_image: Optional[|resource_type|]
+            Update the cover_image for rich presence integrations.
+        guild_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The guild installation type configuration for this application.
+            If set to ``None``, guild installations will be disabled.
+        user_install_type_config: Optional[:class:`InstallTypeConfiguration`]
+            The user installation type configuration for this application.
+            If set to ``None``, user installations will be disabled.
+        event_webhooks_url: Optional[:class:`str`]
+            The application's event webhooks URL.
+        event_webhooks_status: :class:`ApplicationEventWebhookStatus`
+            The application's event webhooks status.
+        event_webhook_types: Optional[List]
+            The application's event webhook types. See `webhook event types <https://discord.com/developers/docs/events/webhook-events#event-types>`_
+            for a list of valid events.
+
+        Raises
+        ------
+        HTTPException
+            Editing the application information failed.
+
+        Returns
+        -------
+        :class:`.AppInfo`
+            The new application information.
+        """
+        fields: EditAppInfoPayload = {}
+        if guild_install_type_config is not MISSING or user_install_type_config is not MISSING:
+            integration_types_config: Dict[str, ApplicationIntegrationTypeConfigurationPayload] = {}
+
+            if guild_install_type_config is not None:
+                if guild_install_type_config is MISSING:
+                    guild_install_type_config = self.guild_install_type_config
+                if guild_install_type_config:
+                    integration_types_config["0"] = guild_install_type_config.to_dict()
+
+            if user_install_type_config is not None:
+                if user_install_type_config is MISSING:
+                    user_install_type_config = self.user_install_type_config
+                if user_install_type_config:
+                    integration_types_config["1"] = user_install_type_config.to_dict()
```

Comment:
I don't think the `is not None` check is necessary, it's already a logical consequence of both if's
```suggestion
            if guild_install_type_config is MISSING:
                guild_install_type_config = self.guild_install_type_config
            if guild_install_type_config:
                integration_types_config["0"] = guild_install_type_config.to_dict()

            if user_install_type_config is MISSING:
                user_install_type_config = self.user_install_type_config
            if user_install_type_config:
                integration_types_config["1"] = user_install_type_config.to_dict()
```

Reply:
https://github.com/DisnakeDev/disnake/pull/1237/commits/fa9d1b4b67d8a7682d2d4e1d6dca38f21640aeaa